src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -0,0 +1,513 @@
+            PathValidationData pathValidationData) {
+        List<BiFunction<InputData, RepositoryData, Set<String>>> validationFunctions = new LinkedList<>();
+
+        validationFunctions.add(this::validateForwardAndReverseLinks);
+
+        if (isLatencyValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateLatency);
+        }
+
+        if (isLatencyTier2ValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateLatencyTier2);
+        }
+
+        if (isBandwidthValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateBandwidth);
+        }
+
+        if (isDiverseWithFlowValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateDiverseWithFlow);
+        }
+
+        if (isEncapsulationTypeValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateEncapsulationType);
+        }
+
+        return validationFunctions;
+    }
+
+    private boolean isEncapsulationTypeValidationRequired(PathValidationData pathValidationData) {
+        return pathValidationData.getFlowEncapsulationType() != null;
+    }
+
+    private boolean isDiverseWithFlowValidationRequired(PathValidationData pathValidationData) {
+        return StringUtils.isNotBlank(pathValidationData.getDiverseWithFlow());
+    }
+
+    private boolean isBandwidthValidationRequired(PathValidationData pathValidationData) {
+        return pathValidationData.getBandwidth() != null
+                && pathValidationData.getBandwidth() != 0
+                && (pathValidationData.getPathComputationStrategy() == null
+                || pathValidationData.getPathComputationStrategy() == COST_AND_AVAILABLE_BANDWIDTH);
+    }
+
+    private boolean isLatencyTier2ValidationRequired(PathValidationData pathValidationData) {
+        return pathValidationData.getLatencyTier2() != null
+                && !pathValidationData.getLatencyTier2().isZero()
+                && (pathValidationData.getPathComputationStrategy() == null
+                || pathValidationData.getPathComputationStrategy() == PathComputationStrategy.LATENCY
+                || pathValidationData.getPathComputationStrategy() == MAX_LATENCY);
+    }
+
+    private boolean isLatencyValidationRequired(PathValidationData pathValidationData) {
+        return pathValidationData.getLatency() != null
+                && !pathValidationData.getLatency().isZero()
+                && (pathValidationData.getPathComputationStrategy() == null
+                || pathValidationData.getPathComputationStrategy() == PathComputationStrategy.LATENCY
+                || pathValidationData.getPathComputationStrategy() == MAX_LATENCY);
+    }
+
+    private Set<String> validateForwardAndReverseLinks(InputData inputData, RepositoryData repositoryData) {
+        Map<SwitchId, Switch> switchMap = repositoryData.getSwitchRepository().findByIds(
```

Comment:
why can't we use switch repository from `this`?

Reply:
We could. In the beginning, I made these methods without dependencies, so that we can move them easily to other utility classes if this class grows too big and I wasn't sure if we want to have them static and if I want to use repos all the time or use some caching, or hide where the actual data came from (make methods like findByIds right in repositoryData and hide where it came from). So I thought that from this point it is easier to go in any direction.
What's your recommendation here?

Reply:
I would recommend to use this.repository just to do not make current code more complicated. If we will need to move it to some other class - we will path repos inside

Reply:
done